reccmp/isledecomp/formats/pe.py

```diff
@@ -0,0 +1,1005 @@
+    IMAGE_SCN_LNK_REMOVE = 0x00000800
+    IMAGE_SCN_LNK_COMDAT = 0x00001000
+    IMAGE_SCN_GPREL = 0x00008000
+    IMAGE_SCN_MEM_PURGEABLE = 0x00020000
+    IMAGE_SCN_MEM_16BIT = 0x00020000
+    IMAGE_SCN_MEM_LOCKED = 0x00040000
+    IMAGE_SCN_MEM_PRELOAD = 0x00080000
+    # IMAGE_SCN_ALIGN_1BYTES = 0x00100000
+    # IMAGE_SCN_ALIGN_2BYTES = 0x00200000
+    # IMAGE_SCN_ALIGN_4BYTES = 0x00300000
+    # IMAGE_SCN_ALIGN_8BYTES = 0x00400000
+    # IMAGE_SCN_ALIGN_16BYTES = 0x00500000
+    # IMAGE_SCN_ALIGN_32BYTES = 0x00600000
+    # IMAGE_SCN_ALIGN_64BYTES = 0x00700000
+    # IMAGE_SCN_ALIGN_128BYTES = 0x00800000
+    # IMAGE_SCN_ALIGN_256BYTES = 0x00900000
+    # IMAGE_SCN_ALIGN_512BYTES = 0x00A00000
+    # IMAGE_SCN_ALIGN_1024BYTES = 0x00B00000
+    # IMAGE_SCN_ALIGN_2048BYTES = 0x00C00000
+    # IMAGE_SCN_ALIGN_4096BYTES = 0x00D00000
+    # IMAGE_SCN_ALIGN_8192BYTES = 0x00E00000
+    IMAGE_SCN_LNK_NRELOC_OVFL = 0x01000000
+    IMAGE_SCN_MEM_DISCARDABLE = 0x02000000
+    IMAGE_SCN_MEM_NOT_CACHED = 0x04000000
+    IMAGE_SCN_MEM_NOT_PAGED = 0x08000000
+    IMAGE_SCN_MEM_SHARED = 0x10000000
+    IMAGE_SCN_MEM_EXECUTE = 0x20000000
+    IMAGE_SCN_MEM_READ = 0x40000000
+    IMAGE_SCN_MEM_WRITE = 0x80000000
+
+
+@dataclasses.dataclass(frozen=True)
+class PEImageSectionHeader:
+    # Order is significant!
+    name: str
+    virtual_size: int
+    virtual_address: int
+    size_of_raw_data: int
+    pointer_to_raw_data: int
+    pointer_to_relocations: int
+    pointer_to_line_numbers: int
+    number_of_relocations: int
+    number_of_line_numbers: PESectionFlags
+
+    @classmethod
+    def from_memory(
+        cls, data: bytes, offset: int, count: int
+    ) -> tuple[tuple["PEImageSectionHeader", ...], int]:
+        struct_fmt = "<8s8I"
+        s_size = struct.calcsize(struct_fmt)
+        items = tuple(
+            cls(
+                members[0].decode("ascii").rstrip("\x00"),
+                *members[1:-1],
+                PESectionFlags(members[-1]),
+            )
+            for members in struct.iter_unpack(
+                struct_fmt, data[offset : offset + count * s_size]
+            )
+        )
+        return items, offset + count * struct.calcsize(struct_fmt)
+
+
+@dataclasses.dataclass
+class CodeViewHeaderNB10:
+    cv_signature: bytes  # "NB10" (or NBxx?)
+    offset: int  # always 0 for NB20
+    signature: int  # seconds since 1970-01-01
+    age: int  # incrementing value
+    pdb_file_name: bytes  # zero terminated string with the name of the PDB file
+
+    CV_SIGNATURE = b"NB10"
+
+    @classmethod
+    def from_memory(cls, data: bytes, offset: int) -> Optional["CodeViewHeaderNB10"]:
+        struct_fmt = "<4sIII"
+        if not cls.taste(data, offset):
+            raise ValueError
+        items = struct.unpack_from(struct_fmt, data, offset)
+        offset_pdb_filename = offset + struct.calcsize(struct_fmt)
+        pos_null = data.index(0, offset_pdb_filename)
+        if pos_null == -1:
```

Comment:
`bytes.index()` throws `ValueError` if not found. Should use `find()` instead or wrap in try/catch. Same for the RSDS header